filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala

```diff
@@ -15,9 +15,24 @@ case class DrsPathBuilder(fileSystemProvider: DrsCloudNioFileSystemProvider) ext
 
   override def name: String = "DRS"
 
+  /**
+    * Unlike other cloud providers where directories are identified with a trailing slash at the end like `gs://bucket/dir/`,
+    * DRS has a concept of bundles for directories (not supported yet). This sanitizes the path by removing the trailing '/'(s)
+    * so that the CloudNioFileSystemProvider & CloudNioPath does not treat a DRS path ending with '/' as a directory, otherwise
+    * it's size is returned as 0
+    */
+  private def removeTrailingSlashes(path: String): String = {
+    if (path.length > (drsScheme.length + 3)) { //3: length of '://'
+      val pathArray = path.split(s"://")
+      val transformedPath = pathArray(1).replaceAll("[/]+$", "")
+      s"$drsScheme://$transformedPath"
+    }
+    else path
+  }
+
   override def build(pathAsString: String): Try[Path] = {
     if (pathAsString.startsWith(s"$drsScheme://")) {
-      Try(URI.create(UrlEscapers.urlFragmentEscaper().escape(pathAsString))) flatMap { uri =>
+      Try(URI.create(UrlEscapers.urlFragmentEscaper().escape(removeTrailingSlashes(pathAsString)))) flatMap { uri =>
```

Comment:
This line is a bit busy, I'd like to see a variable introduced, i.e.
```suggestion
      val trailingSlashedRemoved = removeTrailingSlashes(pathAsString)
      Try(URI.create(UrlEscapers.urlFragmentEscaper().escape(trailingSlashesRemoved)) flatMap { uri =>
```

Reply:
Note that if the new code can throw exceptions (and currently it can) then we should introduce a for comprehension.

```suggestion
      val trailingSlashedRemoved = removeTrailingSlashes(pathAsString)

      def checkScheme(uri: URI) =
        if (!Option(uri.getScheme).exists(_.equalsIgnoreCase(fileSystemProvider.getScheme))) {
          Failure(new IllegalArgumentException(s"$pathAsString does not have a $drsScheme scheme."))
        } else Try(())

      for {
        escapedUrl <- Try(UrlEscapers.urlFragmentEscaper().escape(trailingSlashedRemoved))
        uri <- Try(URI.create(escapedUrl))
        _ <- checkScheme(uri)
        path <- Try(DrsPath(fileSystemProvider.getPath(uri)))
      } yield path
```
I know this is outside of the scope of the problem, but the code is getting to the point where we're losing control of maintainability, so it's worth it.